src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.State.cs

```diff
@@ -265,16 +349,278 @@ private static bool IsValidAttributeParameterType(ITypeSymbol type)
             }
 
             private async Task<bool> TryDetermineTypeToGenerateInAsync(
-                SemanticDocument document,
-                INamedTypeSymbol original,
-                CancellationToken cancellationToken)
+                INamedTypeSymbol original, CancellationToken cancellationToken)
             {
-                var definition = await SymbolFinder.FindSourceDefinitionAsync(original, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+                var definition = await SymbolFinder.FindSourceDefinitionAsync(original, _document.Project.Solution, cancellationToken).ConfigureAwait(false);
                 TypeToGenerateIn = definition as INamedTypeSymbol;
 
-                return TypeToGenerateIn != null &&
-                    (TypeToGenerateIn.TypeKind == TypeKind.Class ||
-                     TypeToGenerateIn.TypeKind == TypeKind.Struct);
+                return TypeToGenerateIn?.TypeKind == TypeKind.Class || TypeToGenerateIn?.TypeKind == TypeKind.Struct;
+            }
+
+            private void GetParameters(
+                ImmutableArray<TArgumentSyntax> arguments,
+                ImmutableArray<TAttributeArgumentSyntax>? attributeArguments,
+                ImmutableArray<ITypeSymbol> parameterTypes,
+                ImmutableArray<ParameterName> parameterNames,
+                CancellationToken cancellationToken)
+            {
+                var parameterToExistingMemberMap = ImmutableDictionary.CreateBuilder<string, ISymbol>();
+                var parameterToNewFieldMap = ImmutableDictionary.CreateBuilder<string, string>();
+                var parameterToNewPropertyMap = ImmutableDictionary.CreateBuilder<string, string>();
+
+                using var _ = ArrayBuilder<IParameterSymbol>.GetInstance(out var parameters);
+
+                for (var i = 0; i < parameterNames.Length; i++)
+                {
+                    // See if there's a matching field or property we can use.  First test in a case sensitive
+                    // manner, then case insensitively.
+                    if (!TryFindMatchingFieldOrProperty(
```

Comment:
Any reason not to just combine the `caseSensitive` `true`/`false` cases inside of `TryFindMatchingFieldOrProperty`? Feels weird that we always need to call this method twice with both argument combinations to get the correct behavior

Reply:
fair.  let me see about that!

Reply:
done!